experimental/packages/opentelemetry-sdk-metrics-base/src/state/MeterProviderSharedState.ts

```diff
@@ -14,14 +14,26 @@
  * limitations under the License.
  */
 
+import { HrTime } from '@opentelemetry/api';
+import { hrTime } from '@opentelemetry/core';
 import { Resource } from '@opentelemetry/resources';
+import { Meter } from '../Meter';
 import { ViewRegistry } from '../view/ViewRegistry';
+import { MetricCollector } from './MetricCollector';
 
 /**
  * An internal record for shared meter provider states.
  */
 export class MeterProviderSharedState {
   viewRegistry = new ViewRegistry();
+  // TODO: we should probably Object.freeze here but,
```

Comment:
i don't think its necessary, typescript will user will have error anyway and we don't freeze in other part of the API/SDK so javascript user can mess up our state without issue already

Reply:
A cast would also work fine I think `Object.freeze(hrTime()) as [number, number]`. In any case, I agree with @vmarchaud since we don't really do it anywhere else either.